dom/ipc/ContentParent.cpp

```diff
@@ -909,11 +909,13 @@ nsString ContentParent::GetRecording(Element* aFrameElement) {
 already_AddRefed<ContentParent> ContentParent::GetUsedBrowserProcess(
     const nsACString& aRemoteType, nsTArray<ContentParent*>& aContentParents,
     uint32_t aMaxContentParents, bool aPreferUsed) {
-  // This code is disabled, as the presence of recording tabs can cause other
-  // new tabs to not render. There might be a simple fix here but it doesn't
-  // seem worth investigating.
-  return nullptr;
-  /*
+  // This code is disabled behind a pref, as the presence of recording tabs
+  // can cause other new tabs to not render. There might be a simple fix
+  // here but it doesn't seem worth investigating.
+  if (!Preferences::GetBool("devtools.recordreplay.preloadProcess")) {
+    return nullptr;
+  }
+
```

Comment:
I'm confused, if we don't bail here, doesn't that mean that pressing record may just reuse an existing non-recording process?

Reply:
That's what I was thinking too.  But I don't know when specifically the recording hooks get set up in the process.  I don't know if the recording browser child process needs to be freshly started to work, or if we hook in after it has started.  Either way, this just puts this logic behind a pref.

From what I've understood, turning this on makes graphics/rendering not work.. as opposed to "nothing works".  So I suspect that getting a recording browser child is something that happens _after_ the browser child process is created?

I don't know.  I've left some messages for Brian in the other issue: https://github.com/RecordReplay/backend/issues/1088